automated/utils/test-runner.py

```diff
@@ -809,17 +812,39 @@ def send_to_qa_reports(self):
         with open("{}/stdout.log".format(self.test['test_path']), "r") as logfile:
             log = logfile.read()
 
-        submit_results(
-            group_project_slug="{}/{}".format(self.qa_reports_group, self.qa_reports_project),
-            build_version=self.qa_reports_build_version,
-            env_slug=self.qa_reports_env,
-            tests=tests,
-            metrics=metrics,
-            log=log,
-            metadata=self.testdef['metadata'],
-            attachments=None,
-        )
-        self.logger.info("Results pushed to QA Reports")
+        metadata = {}
+        if self.qa_reports_metadata:
+            metadata.update(self.qa_reports_metadata)
+        if self.qa_reports_metadata_file:
+            try:
+                with open(self.qa_reports_metadata_file, "r") as metadata_file:
+                    loaded_metadata = yaml.load(metadata_file, Loader=yaml.SafeLoader)
+                    # check if loaded metadata is key=value and both are strings
+                    for key, value in loaded_metadata.items():
+                        if type(key) == str and type(value) == str:
+                            # only update metadata with simple keys
+                            # ignore all other items in the dictionary
+                            metadata.update({key: value})
+                        else:
+                            self.logger.warning("Ignoring key: %s" % key)
+            except FileNotFoundError:
+                self.logger.warning("Metadata file not found")
+            except PermissionError:
+                self.logger.warning("Insufficient permissions to open metadata file")
+        if self.qa_reports_disable_metadata:
```

Comment:
Can this be moved at the to at line 816 and move the top section in else block. This will avoid unnecessary parsing if disable_metadata is taking precedence.

Reply:
Fixed.